presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -242,6 +245,28 @@ def __remove_low_scores(
         new_results = [result for result in results if result.score >= score_threshold]
         return new_results
 
+    def __ensure_recognition_metadata(
+        self, results: List[RecognizerResult], recognizer: EntityRecognizer
+    ):
+        """Ensure recognition metadata with recognizer name existence.
+
+        Ensure recognizer result list contains recognizer name inside recognition
+        metadata dictionary, and if not create it. recognizer_name is needed
+        for context aware enhancement
+
+        :param results: List of RecognizerResult
+        :param recognizer: Entity recognizer
+        """
+        for result in results:
+            if not result.recognition_metadata:
+                result.recognition_metadata = dict()
+            if not result.recognition_metadata.get(
```

Comment:
possibly: if not RecognizerResult.RECOGNIZER_NAME_KEY in result.recognition_metadata

Reply:
thanks, I've changed it to "if RecognizerResult.RECOGNIZER_NAME_KEY not in result.recognition_metadata:" to comply with E713 rule